src/Infrastructure/LeanCode.Firebase.FCM/PushNotificationTokenEntity.cs

```diff
@@ -0,0 +1,24 @@
+using LeanCode.TimeProvider;
+
+namespace LeanCode.Firebase.FCM;
+
+public class PushNotificationTokenEntity<TUserId>
```

Comment:
Can we change this to a record?

Reply:
We could, but what's the point of structural equality for those tokens? `DateCreated` makes comparing instances tricky anyway and we don't actually ever instanciate those entities in memory.

Plus, now that we are actually updating token's users instead of replacing them, perhaps renaming this column to `DateModified` would be more appropriate?

Reply:
In my mind the change was not really related to the structural equality but only to the syntax of this class - will be shorter (and w/o public setters).

Reply:
If we're to do this in name of shorter syntax, sure, I can try that.

What about `DateCreated` though, should it stay exactly as is?

Reply:
Your call. For me, `DateCreated` is perfectly OK. Although we modify the record, the _meaning_ stays the same - when the token itself has been created. `DateModified` would relate more to the record than the token, which also makes sense.